graphql/src/execution/query.rs

```diff
@@ -150,8 +156,16 @@ impl Query {
         max_complexity: Option<u64>,
         max_depth: u8,
     ) -> Result<Arc<Self>, Vec<QueryExecutionError>> {
-        let validation_errors =
-            validate(schema.document(), &query.document, &GRAPHQL_VALIDATION_PLAN);
+        let mut validation_cache = GRAPHQL_VALIDATION_CACHE
+            .lock()
+            .unwrap_or_else(PoisonError::into_inner);
+        let validation_errors = validation_cache.entry(query.shape_hash).or_insert_with(|| {
+            validate(
+                &schema.document(),
+                &query.document,
+                &GRAPHQL_VALIDATION_PLAN,
+            )
+        });
```

Comment:
It seems the lock is held for a larger scope than necessary, maybe you can do it all in one line:
```suggestion
        GRAPHQL_VALIDATION_CACHE
            .lock()
            .unwrap_or_else(PoisonError::into_inner)
            .entry(query.shape_hash).or_insert_with(|| {
            validate(
                &schema.document(),
                &query.document,
                &GRAPHQL_VALIDATION_PLAN,
            )
        });
```

Reply:
@leoyvens not sure how to do it in one line and assign the result to `let validation_errors`